### ISSUE TITLE
config/tls directory is not read when not listening on SMTPS port

### ISSUE THREAD

**Issue by Juerd** [haraka/Haraka#2905]
**Describe the bug**
When the `smtps_port` is not also specified in `listen` (which could be deliberate), `get_certs_dir` is not called, causing the certificates in the `config/tls` directory to be ignored for STARTTLS sessions. They do get loaded, and do work with STARTTLS, if the SMTPS port is added to `listen`.

**Steps To Reproduce**
1. Example `config/smtp.ini`
2. Add certificates in `config/tls`
3. Observe that the message `[INFO] [-] [core] found {} TLS certs in config/tls` is missing from the log

**System Info:**

Haraka | Haraka.js — Version: 2.8.27
 --- | :--- 
Node | v10.23.1
OS | Linux (hostname censored) 4.19.0-13-amd64 #1 SMP Debian 4.19.160-2 (2020-11-28) x86_64 GNU/Linux
openssl | OpenSSL 1.1.1d  10 Sep 2019

**Comment by msimerson** [comment]
Indeed. 

Haraka provides encryption via the TLS plugin. That encapsulation has introduced additional complexity and most of the TLS code has moved out of the plugin. As the importance of authentication, authorization, and encryption have increased, we no longer consider TLS optional. Also, node's TLS support has evolved considerably. We will likely kill off the TLS plugin to further simplify our TLS handling. In the meantime, I think this patch will address your issue, at the risk of failing badly if TLS is not configured properly.

```diff
diff --git a/server.js b/server.js
index f79d9130..84800369 100644
--- a/server.js
+++ b/server.js
@@ -397,8 +397,10 @@ Server.get_smtp_server = (ep, inactivity_timeout, done) => {
     else {
         server = tls_socket.createServer(onConnect);
         server.has_tls = false;
-        Server.listeners.push(server);
-        done(server);
+        tls_socket.getSocketOpts('*', opts => {
+            Server.listeners.push(server);
+            done(server);
+        })
     }
 }
 
```

**Comment by Juerd** [comment]
Thanks. I've already worked around it by not having port 465 open in the firewall. 

I understand that SMTPS is not enabled in the example configuration, since it's deprecated, but may I suggest adding a warning note to the documentation of the `config/tls` directory until the functionality works on starttls-only installations?

**Comment by msimerson** [comment]
> I understand that SMTPS is not enabled in the example configuration, since it's deprecated,

Not quite. It's not enabled in the example because port 465 is only used when acting as a MSA. While port 465 was deprecated, it has since been un-deprecated, as of RFC some thing or other.

> adding a warning note to the documentation of the config/tls directory

😬 I'd rather make it work as users of Haraka would expect.
